### DISCUSSION TITLE
2.0.0 - Beta 1

### DISCUSSION THREAD

**Discussion by jarednova** [timber/timber#2625] [General]
This is the first beta version of the long awaited Version 2 of Timber. We consider this beta feature-complete. We will concentrate on fixing bugs to be able to soon release a first release candidate.

There are quite a few breaking changes in Version 2 of Timber. Refer to the [Upgrade Guide](https://timber.github.io/docs/v2/upgrade-guides/2.0/) for more information.

### The overall goals of Timber 2.0 include:
- Making Timber more consistent.
- Making Timber easier to handle and extend.
- Refactoring how Timber Core works under the hood to improve compatibility with WordPress Core and be ready for future challenges.
- Making Timber more compatible with other plugins.

### High-level changes include:
- Timber will only work as a Composer package.
- A newer, streamlined API for accessing [Posts](https://timber.github.io/docs/v2/guides/posts/), [Terms](https://timber.github.io/docs/v2/guides/terms/), [Users](https://timber.github.io/docs/v2/guides/users/), [Comments](https://timber.github.io/docs/v2/guides/comments/) and [Menus](https://timber.github.io/docs/v2/guides/menus/).
- [Class Maps](https://timber.github.io/docs/v2/guides/class-maps/) for a more loosely coupled way to [extend Timber](https://timber.github.io/docs/v2/guides/extending-timber/) with your own Post, Term, User, Menu, MenuItem, and Comment objects.
- No more direct instantiation of the classes mentioned above. Use [Class Maps](https://timber.github.io/docs/v2/guides/class-maps/) instead.
- New [PostCollectionInterface](https://timber.github.io/docs/v2/guides/posts#post-collections) for a unified way to deal with various lists of posts.
- An upgraded [Context](https://timber.github.io/docs/v2/upgrade-guides/2.0/#context).
- A big update for how fetching [meta values](https://timber.github.io/docs/v2/upgrade-guides/2.0/#meta) works.
- An updated [WP-CLI integration]().
- Better options to control and [extend Twig](https://timber.github.io/docs/v2/guides/extending-twig/).
- Better integration with the WordPress [Date and Time functionality](https://timber.github.io/docs/v2/guides/date-time/).

We’re happy if you want to test out the first Timber beta. In case you find errors, please [open an issue](https://github.com/timber/timber/issues) (and label with `2.x` and `bug`) or [create a discussion](https://github.com/timber/timber/discussions) in case you have questions.

We want to thank so so so many people for being a part of Timber and helping us achieve this milestone. We're so proud of the work and the evolution to a stable and widely-used part of so many developer's tools.

----

If you've been using the alpha versions, here's what's changed in this version of `1.0.0-beta.1`:
 

### What's Changed
* Add a .git-blame-ignore-revs file by @gchtr in https://github.com/timber/timber/pull/2606
* 2.x Add missing query() method to `Timber\PostQuery` by @gchtr in https://github.com/timber/timber/pull/2613
* 2.x Reintroduce Timber’s WP-CLI integration by @gchtr in https://github.com/timber/timber/pull/2611
* 2.x Remove unused WPML integration class by @gchtr in https://github.com/timber/timber/pull/2616
* 2.x Remove `Attachment::$caption` property in favor of `Attachment::caption()` method by @gchtr in https://github.com/timber/timber/pull/2612
* Make snippets CS compliant by @nlemoine in https://github.com/timber/timber/pull/2620
* 2.x update tests fix indentation by @nlemoine in https://github.com/timber/timber/pull/2622
* 2.x Fix tests in #2614 by @gchtr in https://github.com/timber/timber/pull/2621
* Use `yoast/wp-test-utils` by @nlemoine in https://github.com/timber/timber/pull/2614
* 2.x Document the `timber/menuitem/classmap` filter by @gchtr in https://github.com/timber/timber/pull/2618
* 2.x Add wp_object() method and $wp_object property for Core objects by @gchtr in https://github.com/timber/timber/pull/2615
* 2.x Update Upgrade Guide by @gchtr in https://github.com/timber/timber/pull/2617
* 2x Update CONTRIBUTING.md by @gchtr in https://github.com/timber/timber/pull/2619

**Full Changelog**: https://github.com/timber/timber/compare/2.0.0-alpha.5...2.0.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/timber/timber/releases/tag/2.0.0-beta.1'>2.0.0 - Beta 1</a>.</em>

**Comment by acobster** [comment]
This is huge!!! Congrats on an awesome looking Beta, guys! 😁🚀

**Reply by gchtr** [reply]
@acobster We wouldn’t be here without you, Coby. A huge thank you for all your work! 💪

**Reply by jarednova** [reply]
Indeed, hope things are going good with you @acobster !!!!

**Reply by begroff** [reply]
I noticed the assets/downloading the composer package doesn't have a composer.json file. Is this by design?

**Comment by Levdbas** [comment]
Hi everyone, the team is hard at work to make Timber 2.0 happen. We've just created an updated roadmap with open tickets, pull requests, etc. If you would like to help out on Timber, now is the perfect timing. See https://github.com/timber/timber/issues/2741

Closing this discussion in favor of the mentioned ticket above.
